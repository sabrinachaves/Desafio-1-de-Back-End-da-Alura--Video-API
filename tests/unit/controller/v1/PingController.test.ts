import { Request } from 'express';
import PingController from '../../../../src/application/v1/controller/PingController';

describe('PingController', () => {
  it('should handle the request #unit', async () => {
    const instance = new PingController();
    const req = {} as Request;
    const mockJsonResponse = jest.fn();
    const res: any = {
      status: jest.fn().mockImplementation(() => ({
        json: mockJsonResponse,
      })),
    };

    instance.execute(req, res);
    expect(res.status).toHaveBeenCalled();
    expect(res.status).toHaveBeenCalledWith(200);
    expect(mockJsonResponse).toHaveBeenCalled();
  });

  it('should match the snapshot #unit', () => {
    const instance = new PingController();
    expect(instance).toMatchSnapshot();
  });
});
