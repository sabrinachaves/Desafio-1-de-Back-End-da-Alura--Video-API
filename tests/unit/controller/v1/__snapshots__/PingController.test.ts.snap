// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PingController should match the snapshot #unit 1`] = `
PingController {
  "execute": [Function],
}
`;
